metrics.nim

```diff
@@ -478,15 +463,14 @@ proc incGauge(gauge: Gauge, amount: int64|float64 = 1, labelValues: LabelsParam
     try:
       var timestamp = getTime().toMilliseconds()
 
-      let labelValuesCopy = validateGaugeLabelValues(gauge, labelValues)
-
-      atomicAdd(gauge.metrics[labelValuesCopy][0].value.addr, amount.float64)
-      atomicStore(cast[ptr int64](gauge.metrics[labelValuesCopy][0].timestamp.addr), timestamp.addr, ATOMIC_SEQ_CST)
-
-      pushMetrics(name = gauge.name,
-                  value = gauge.metrics[labelValuesCopy][0].value,
-                  metricType = "g",
-                  timestamp = timestamp)
+      withLock gauge.lock:
+        let labelValuesCopy = validateGaugeLabelValues(gauge, labelValues)
+        gauge.metrics[labelValuesCopy][0].value += amount.float64
+        gauge.metrics[labelValuesCopy][0].timestamp = timestamp
+        pushMetrics(name = gauge.name,
```

Comment:
if one would copy the value out of the lock, pushMetrics could be done without locking which probably is prudent in a truly multithreaded app with multiple consumers - this can be left for a future pr however

Reply:
Not worth the trouble. `pushMetrics()` is a very fast, "non-locking" operation that tries to put the payload in a channel's send buffer and simply gives up if that buffer is full.